client/src/app.js

```diff
@@ -0,0 +1,94 @@
+import Component from "./core/component.js";
+import Header from "./components/Header/header.js"
+import MainPage from "./components/mainPage.js";
+import { initialRoutes } from './router.js';
+import SubPage from "./components/SubPages/subPage.js";
+import FavoritePage from "./components/Favorite/favoritePage.js";
+
+export default class App extends Component{
+
+  template(){
+    return `
+      <Header data-component="Header"></Header>
+      <div id="page"></div>
+      `;
+  }
+
+  mount(){
+    
+    const pageDiv=this.$target.querySelector("#page");
+    initialRoutes(pageDiv);
+
+    const $header=this.$target.querySelector('[data-component="Header"]');
+    const $mainPage=this.$target.querySelector('[data-component="MainPage"]');
+    const $subPage=this.$target.querySelector('[data-component="SubPage"]');
```

Comment:
```suggestion
    const pageDiv=this.$target.querySelector("#page");
    initialRoutes(pageDiv);

    const $header = this.$target.querySelector('[data-component="Header"]');
    const $mainPage = this.$target.querySelector('[data-component="MainPage"]');
    const $subPage = this.$target.querySelector('[data-component="SubPage"]');
```
가독성을 위해서 여백을 주면 더 좋을 것 같습니다.